tests/accumulate-headers-after-304.js

```diff
@@ -0,0 +1,178 @@
+            testCase.client().request.conditions({ ims: resource.notModifiedSince() });
+            testCase.client().checks.add((client) => {
+                client.expectStatusCode(304);
+            });
+            await testCase.run();
+        }
+
+        let updatingResponse = null;
+        {
+            let testCase = new HttpTestCase(`get a 304 that increases the cached header size on ${updateHeaderLength}`);
+
+            resource.modifyNow();
+            resource.expireAt(FuzzyTime.DistantFuture());
+
+            testCase.client().request.for(resource);
+            testCase.client().request.conditions({ ims: resource.modifiedSince() });
+            testCase.client().request.header.add("Cache-Control", "max-age=0");
+            if (Config.smp())
+                testCase.client().nextHopAddress = this._workerListeningAddresses[2];
+
+            testCase.server().response.header.add(updateField);
+            testCase.server().response.startLine.code(304);
+            testCase.server().serve(resource);
+            testCase.check(() => {
+                const receivedResponse = testCase.client().transaction().response;
+                const code = receivedResponse.startLine.codeInteger();
+                updatingResponse = testCase.server().transaction().response;
+                if (headerSize <= MaxHeaderSize) {
+                    assert.equal(updatingResponse.id(), receivedResponse.id(), "relayed X-Daft-Response-ID");
+                    assert.equal(code, 200);
+                } else {
+                    assert(code>=500);
+                }
+            });
+            await testCase.run();
+        }
+
+        {
+            let testCase = new HttpTestCase('hit updated headers');
+            testCase.client().request.for(resource);
+            if (Config.smp())
+                testCase.client().nextHopAddress = this._workerListeningAddresses[3];
+            testCase.check(() => {
+                let updatedResponse = testCase.client().transaction().response;
+                const code = updatedResponse.startLine.codeInteger();
+                if (headerSize <= MaxHeaderSize) {
+                    assert.equal(code, 200);
+                    assert.equal(updatingResponse.id(), updatedResponse.id(), "updated X-Daft-Response-ID");
+                    assert.equal(updatedResponse.header.values("Last-Modified"), resource.lastModificationTime.toUTCString(), "updated Last-Modified");
+                    assert.equal(updatedResponse.header.value(hitCheck.name), hitCheck.value, "preserved originally cached header field");
+                } else {
+                    assert(code>=500);
+                }
+            });
+            await testCase.run();
+        }
+
+        AddressPool.ReleaseListeningAddress(resource.uri.address);
+    }
+
+    async run(/*testRun*/) {
+        // TODO: test some other sizes?
+        await this.testOne(MaxHeaderSize);
```

Comment:
I am not sure, but it feels like we can write a single test case that checks everything we want to check (for a given maximum size) rather than writing two test cases (one that does not exceed the maximum and one that does). That single test case would first successfully cache the maximum-size header and then update that header to exceed the maximum.

We already have a test that checks whether the headers are successfully updated (when they can be) -- proxy-update-headers-after-304.js. This new test does not need to repeat those existing checks IMO.